libs/api/deployment/src/lib/core/service/deployment-assignment.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Inject, Injectable } from '@nestjs/common';
+
+import { DbSessionProvider } from '@kordis/api/shared';
+
+import {
+	DEPLOYMENT_ASSIGNMENT_REPOSITORY,
+	DeploymentAssignmentRepository,
+} from '../repository/deployment-assignment.repository';
+import {
+	UNIT_ASSIGNMENT_REPOSITORY,
+	UnitAssignmentRepository,
+} from '../repository/unit-assignment.repository';
+
+@Injectable()
+export class DeploymentAssignmentService {
+	constructor(
+		@Inject(DEPLOYMENT_ASSIGNMENT_REPOSITORY)
+		private readonly deploymentAssignmentRepository: DeploymentAssignmentRepository,
+		@Inject(UNIT_ASSIGNMENT_REPOSITORY)
+		private readonly unitAssignmentRepository: UnitAssignmentRepository,
+	) {}
+
+	/**
+	 * Sets the assignments of a deployment by completely removing all previous assignments and assigning the new units and alert groups.
+	 * Keeps units of alert groups that are not assigned to the deployment in their old deployment but without the alert group assignment.
+	 * @param orgId The organization id.
+	 * @param deploymentId The deployment id to assign the units and alert groups to.
+	 * @param unitIds The unit ids to assign to the deployment.
+	 * @param alertGroups The alert groups with their respective unit ids to assign to the deployment.
+	 * @param uow An optional unit of work.
+	 */
+	async setAssignmentsOfDeployment(
+		orgId: string,
+		deploymentId: string,
+		unitIds: string[],
+		alertGroups: {
+			alertGroupId: string;
+			unitIds: string[];
+		}[],
+		uow?: DbSessionProvider | undefined,
+	): Promise<void> {
+		const alertGroupIds: string[] = []; // alert group ids to assign
+		const alertGroupUnitIds: string[][] = []; // alert group unit ids to assign, 1:1 index mapping with alertGroupIds
+		const flatAlertGroupUnitIds: string[] = []; // a flat list of all unit ids of all alert groups
+		for (const alertGroup of alertGroups) {
```

Comment:
I dont, but this is something we need to discuss in general: Should references to other domain entities be checked, and also what does a check mean. The existence ? The correct format of an ID? I think it is not trivial as it would mean we need to query other domains.